### ISSUE TITLE
[BUG] Unexpected value. Please add a @NgModule annotation.

### ISSUE THREAD

**Issue by jdavidls** [angular/angular#33722]
# 🐞 Bug report

### Command (mark with an `x`)

```
- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?

Not, this is a older problem, i reported previusly in angular/angular-cli#13006 without providing a repo.

### Description

 I receive a Unexpected value 'LayoutModule' error when building a library with a single NgModule dependency (one NgModule depends on other).

## 🔬 Minimal Reproduction

This repo contains a minimal example of the issue: https://github.com/prodisign/bug-reporting
`ng build rh`

## 🔥 Exception or Error
<pre><code>
------------------------------------------------------------------------------
Building entry point '@prodisign/rh/model/client'
------------------------------------------------------------------------------
Compiling TypeScript sources through ngc
ERROR: Unexpected value 'LayoutModule in /home/jdavidls/Proyectos/prodisign/experiments/prodisign-workspace/dist/prodisign/rh/client/prodisign-rh-client.d.ts' imported by the module 'ModelModule in /home/jdavidls/Proyectos/prodisign/experiments/prodisign-workspace/projects/prodisign/rh/model/client/src/ModelModule.ts'. Please add a @NgModule annotation.

An unhandled exception occurred: Unexpected value 'LayoutModule in /home/jdavidls/Proyectos/prodisign/experiments/prodisign-workspace/dist/prodisign/rh/client/prodisign-rh-client.d.ts' imported by the module 'ModelModule in /home/jdavidls/Proyectos/prodisign/experiments/prodisign-workspace/projects/prodisign/rh/model/client/src/ModelModule.ts'. Please add a @NgModule annotation.

See "/tmp/ng-1Uj8ei/angular-errors.log" for further details.
</code></pre>

## 🌍 Your Environment
<pre><code>
Angular CLI: 8.3.18
Node: 10.15.3
OS: linux x64
Angular: 8.2.13
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.803.18
@angular-devkit/build-angular      0.803.18
@angular-devkit/build-ng-packagr   0.803.18
@angular-devkit/build-optimizer    0.803.18
@angular-devkit/build-webpack      0.803.18
@angular-devkit/core               8.3.18
@angular-devkit/schematics         8.3.18
@angular/cli                       8.3.18
@ngtools/webpack                   8.3.18
@schematics/angular                8.3.18
@schematics/update                 0.803.18
ng-packagr                         *5.7.1*
rxjs                               6.4.0
typescript                         3.5.3
webpack                            4.39.2
</code></pre>

**Edit:**
In the repo, I am using a complex module disposition, doing use of secondary entrypoints. But other simplest dispositions throwns the same error.

Im reported the same issue at https://github.com/ng-packagr/ng-packagr/issues/1444.

**Comment by jdavidls** [comment]
@alan-agius4 Yeah it is working fine.
Tahnks!

**Comment by alan-agius4** [comment]
@IgorMinar i am going to remove the ivy and ngcc labels as this is a bug in the VE metadata bundler.

**Comment by theodorejb** [comment]
I'm also having this issue when I try to upgrade the angular2-ladda library I maintain to Angular 9. It occurs when trying to run the "Undecorated classes with DI migration".

**Comment by richmondgozarin** [comment]
> This seems related to barrel files.
> 
> In `projects/prodisign/rh/client/src/public_api.ts` change `export * from './layout` to `export * from './layout/index';`
> 
> I am moving this to the FW repo as it's a compiler bug.

Still happening on Angular 9, Had to explicity import the index.ts file.

Any Updates?

**Comment by alxhub** [comment]
Confirmed that this _was_ a bug which is fixed in 10.0 - I reran the repro with an updated `ngcc` and it now correctly compiles the project. Closing as fixed!
